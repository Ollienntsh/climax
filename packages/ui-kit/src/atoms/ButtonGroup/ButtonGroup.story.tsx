import React from 'react';
import { storiesOf } from '@storybook/react';

import ButtonGroup, { ButtonGroupItem } from './ButtonGroup';
import Flex from '../Flex';

const items: ButtonGroupItem<number>[] = [
  {
    value: 1,
    label: 'First',
  },
  {
    value: 2,
    label: 'Second',
  },
];

storiesOf('ui-kit/Atoms', module).add('ButtonGroup', () => (
  <Flex padding={20}>
    <ButtonGroup defaultValue={2} items={items} />
  </Flex>
));
