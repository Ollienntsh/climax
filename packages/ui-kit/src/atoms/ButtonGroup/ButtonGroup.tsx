import React from 'react';
import { Radio as AntRadio } from 'antd';
import { RadioChangeEvent } from 'antd/lib/radio';

export interface ButtonGroupItem<T> {
  label: string;
  value: T;
}

export interface ButtonGroupProps<T> {
  defaultValue?: T;
  items: ButtonGroupItem<T>[];
  onChange?(e: RadioChangeEvent): void;
  value?: T;
}

export default function <T>({
  defaultValue,
  items,
  onChange,
  value,
}: ButtonGroupProps<T>) {
  return (
    <AntRadio.Group
      defaultValue={defaultValue}
      onChange={onChange}
      value={value}
      buttonStyle="solid"
    >
      {items.map(({ label, value }, index) => (
        <AntRadio.Button key={index} value={value}>
          {label}
        </AntRadio.Button>
      ))}
    </AntRadio.Group>
  );
}
