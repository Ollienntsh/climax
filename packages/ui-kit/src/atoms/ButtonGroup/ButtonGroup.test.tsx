import React from 'react';
import { mount } from 'enzyme';
import renderer from 'react-test-renderer';
import serializer from 'jest-emotion';
import snapshotDiff from 'snapshot-diff';

import ButtonGroup, { ButtonGroupItem } from './ButtonGroup';

expect.addSnapshotSerializer(serializer);

const items: ButtonGroupItem<number>[] = [
  {
    value: 1,
    label: 'First',
  },
  {
    value: 2,
    label: 'Second',
  },
];

const defaultTree = renderer.create(<ButtonGroup items={[]} />).toJSON();

describe('ButtonGroup', () => {
  it('should render default', () => {
    expect(defaultTree).toMatchSnapshot();
  });
  it('should render with items', () => {
    const tree = renderer.create(<ButtonGroup items={items} />).toJSON();

    expect(snapshotDiff(tree, defaultTree)).toMatchSnapshot();
  });
  it('should render with second item preselected', () => {
    const tree = renderer
      .create(<ButtonGroup items={items} defaultValue={2} />)
      .toJSON();

    expect(snapshotDiff(tree, defaultTree)).toMatchSnapshot();
  });
  it('should trigger callback on selection change', () => {
    const onChangeMock = jest.fn();
    const wrapper = mount(
      <ButtonGroup items={items} onChange={onChangeMock} />,
    );
    const secondButton = wrapper.find('input').at(1);

    secondButton.simulate('change');

    expect(onChangeMock).toHaveBeenCalled();
    wrapper.unmount();
  });
});
