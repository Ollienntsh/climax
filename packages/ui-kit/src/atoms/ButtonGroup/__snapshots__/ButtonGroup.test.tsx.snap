// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ButtonGroup should render default 1`] = `
<div
  className="ant-radio-group ant-radio-group-solid"
/>
`;

exports[`ButtonGroup should render with items 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <div
    className=\\"ant-radio-group ant-radio-group-solid\\"
- >
-   <label
-     className=\\"ant-radio-button-wrapper\\"
-   >
-     <span
-       className=\\"ant-radio-button\\"
-       style={Object {}}
-     >
-       <input
-         checked={false}
-         className=\\"ant-radio-button-input\\"
-         onBlur={[Function onBlur]}
-         onChange={[Function anonymous]}
-         onFocus={[Function onFocus]}
-         type=\\"radio\\"
-         value={1}
  />
-       <span
-         className=\\"ant-radio-button-inner\\"
-       />
-     </span>
-     <span>
-       First
-     </span>
-   </label>
-   <label
-     className=\\"ant-radio-button-wrapper\\"
-   >
-     <span
-       className=\\"ant-radio-button\\"
-       style={Object {}}
-     >
-       <input
-         checked={false}
-         className=\\"ant-radio-button-input\\"
-         onBlur={[Function onBlur]}
-         onChange={[Function anonymous]}
-         onFocus={[Function onFocus]}
-         type=\\"radio\\"
-         value={2}
-       />
-       <span
-         className=\\"ant-radio-button-inner\\"
-       />
-     </span>
-     <span>
-       Second
-     </span>
-   </label>
- </div>"
`;

exports[`ButtonGroup should render with second item preselected 1`] = `
"Snapshot Diff:
- First value
+ Second value

  <div
    className=\\"ant-radio-group ant-radio-group-solid\\"
- >
-   <label
-     className=\\"ant-radio-button-wrapper\\"
-   >
-     <span
-       className=\\"ant-radio-button\\"
-       style={Object {}}
-     >
-       <input
-         checked={false}
-         className=\\"ant-radio-button-input\\"
-         onBlur={[Function onBlur]}
-         onChange={[Function anonymous]}
-         onFocus={[Function onFocus]}
-         type=\\"radio\\"
-         value={1}
  />
-       <span
-         className=\\"ant-radio-button-inner\\"
-       />
-     </span>
-     <span>
-       First
-     </span>
-   </label>
-   <label
-     className=\\"ant-radio-button-wrapper ant-radio-button-wrapper-checked\\"
-   >
-     <span
-       className=\\"ant-radio-button ant-radio-button-checked\\"
-       style={Object {}}
-     >
-       <input
-         checked={true}
-         className=\\"ant-radio-button-input\\"
-         onBlur={[Function onBlur]}
-         onChange={[Function anonymous]}
-         onFocus={[Function onFocus]}
-         type=\\"radio\\"
-         value={2}
-       />
-       <span
-         className=\\"ant-radio-button-inner\\"
-       />
-     </span>
-     <span>
-       Second
-     </span>
-   </label>
- </div>"
`;
